change/@ni-nimble-components-248b2e34-5be4-492b-8512-1e5226e10ca4.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Now that I've have a chance to play with this in storybook, I'm finding our UX decisions cause some clunky side effects. These are just my opinions but could we double check with @fredvisser that we're happy with these?

1. Only icons in the header. Icon headers are less effective than text headers to convey at a glance what the column represents. In our storybook examples I find myself hovering over the header to see the title. It's more annoying for columns where we don't have a great icon ("Is a Simpson") but I imagine that could happen for real app use cases (e.g. several columns displaying similar concepts). I'd lean towards allowing text headers and wider fixed widths (which opens a can of worms for how we/clients compute that width)
2. Narrow column results in lots of horizontal whitespace. In the screenshot below the icons end up really far from other columns. Obviously setting a smaller table width or adding more columns would help so maybe this is just an issue with our contrived example. But in this case I find myself wanting to bring the icon columns closer to the other columns and put the whitespace on the right edge.
4. No feedback in header when column is interactively sorted. I mentioned this one at your demo. I can live with it but it still bugs me so I'm mentioning it again. If we end up allowing icon columns to be wider we could bring back the sort icons when there's space for them.

<img width="972" alt="image" src="https://github.com/ni/nimble/assets/10500124/e5f870db-37d6-4a4e-95a0-124e081efabf">